src/CommonLib/hostfxr_utility.cpp

```diff
@@ -369,3 +342,166 @@ HOSTFXR_UTILITY::SetHostFxrArguments(
     }
     return hr;
 }
+//
+// Invoke where.exe to find the location of dotnet.exe
+// Copies contents of dotnet.exe to a temp file
+// Respects path ordering.
+
+HRESULT
+HOSTFXR_UTILITY::FindDotnetExePath(
+    _In_ ASPNETCORE_CONFIG* pConfig,
+    _Out_ STRU* struDotnetPath)
+{
+    HRESULT             hr = S_OK;
+    STARTUPINFOW        startupInfo = { 0 };
+    PROCESS_INFORMATION processInformation = { 0 };
+    SECURITY_ATTRIBUTES securityAttributes;
+    STRU                struDotnetSubstring;
+    STRU                struDotnetLocationsString;
+    LPWSTR              pwzDotnetName = NULL;
+    DWORD               dwExitCode;
+    DWORD               dwNumBytesRead;
+    DWORD               dwFilePointer;
+    DWORD               dwBinaryType;
+    INT                 index = 0;
+    INT                 prevIndex = 0;
+    BOOL                fResult = FALSE;
+    BOOL                fIsWow64Process;
+    BOOL                fIsCurrentProcess64Bit;
+    BOOL                fFound = FALSE;
+    CHAR                pzFileContents[READ_BUFFER_SIZE];
+    HANDLE              hStdOutReadPipe = NULL;
+    HANDLE              hStdOutWritePipe = NULL;
+
+    securityAttributes.nLength = sizeof(securityAttributes);
+    securityAttributes.lpSecurityDescriptor = NULL;
+    securityAttributes.bInheritHandle = TRUE;
+
+    if (!CreatePipe(&hStdOutReadPipe, &hStdOutWritePipe, &securityAttributes, 0))
+    {
+        hr = HRESULT_FROM_WIN32(GetLastError());
+        goto Finished;
+    }
+    if (!SetHandleInformation(hStdOutReadPipe, HANDLE_FLAG_INHERIT, 0))
+    {
+        hr = ERROR_FILE_INVALID;
+        goto Finished;
+    }
+
+    // Set stdout and error to redirect to the temp file.
+    startupInfo.cb = sizeof(startupInfo);
+    startupInfo.dwFlags |= STARTF_USESTDHANDLES;
+    startupInfo.hStdOutput = hStdOutWritePipe;
+    startupInfo.hStdError = hStdOutWritePipe;
+
+    // CreateProcess requires a mutable string to be passed to commandline
+    // See https://blogs.msdn.microsoft.com/oldnewthing/20090601-00/?p=18083/
+    pwzDotnetName = SysAllocString(L"\"C:\\Windows\\System32\\where.exe\" dotnet.exe");
```

Comment:
Can you do this in azure web apps?

Reply:
@pan-wang how were you verifying the logging changes before?

Reply:
you cannot use hardcoded absolute path. you can just use "where.exe dotnet". OS will resolve the path of where.exe is.
I opened eventviewer to check the log and the stderr log file.